lib/count-users-by-year-improved.rb

```diff
@@ -0,0 +1,35 @@
+require 'zendesk_api'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+
+  config.retry = true
+end
+
+today = Date.today.prev_day
+lastyear = today - 365
+
+#  puts "Total Tickets: #{@client.users.count}"
+
+puts "Total Users 2012"
+puts @client.search(:query => "type:user created_at>2012-01-01 created_at<2013-01-01 role:end-user -name:Zendesk organization_id:none").count
+puts "Total Users 2013"
+puts @client.search(:query => "type:user created_at>2013-01-01 created_at<2014-01-01 role:end-user -name:Zendesk organization_id:none").count
```

Comment:
It looks like we're going to miss all users created on new year's day. And 2017-04-01 and 2017-07-01.

Reply:
Turns out this is a datetime field so we'd probably only be missing stuff created at exactly 00:00:00